packages/database/src/__tests__/modelClasses/DashboardRelation.test.js

```diff
@@ -0,0 +1,256 @@
+      {
+        dashboard_id: dashboardId,
+        child_id: dashboardItem.id,
+        attributes_filter: filters,
+      },
+      BASE_DASHBOARD_RELATION,
+    );
+  };
+
+  let testEntity;
+  let dashboards;
+  let dashboardIds;
+  let dashboardItems;
+
+  beforeAll(async () => {
+    const entityResult = await findOrCreateDummyCountryEntity(models, {
+      code: 'test',
+      name: 'Test Entity',
+      attributes: { test: 'Yes' },
+    });
+
+    testEntity = entityResult.entity;
+
+    dashboards = await Promise.all(
+      DASHBOARDS.map(dashboard =>
+        findOrCreateDummyRecord(models.dashboard, {
+          ...dashboard,
+          root_entity_code: testEntity.code,
+        }),
+      ),
+    );
+    dashboardIds = dashboards.map(d => d.id);
+
+    dashboardItems = await Promise.all(
+      DASHBOARD_ITEMS.map(dashboardItem => addDashboardItems(dashboardItem)),
+    );
+  });
+
+  afterEach(async () => {
+    await models.dashboardRelation.delete({ dashboard_id: dashboardIds });
+  });
+
+  describe('findDashboardRelationsForEntityAndProject', () => {
+    describe('Filtering', () => {
+      it('should return all dashboard relations for entity and project when there are no filters on any of the relations', async () => {
+        // Add dashboard relations
+        await Promise.all(
+          dashboardItems.map((dashboardItem, i) => addRelation(dashboards[i].id, dashboardItem)),
+        );
+        const result = await models.dashboardRelation.findDashboardRelationsForEntityAndProject(
+          dashboardIds,
+          testEntity.code,
+          'testProject',
+        );
+
+        expect(result).toHaveLength(2);
+      });
+
+      it('should return dashboard items that match entity and project when dashboard relation filters are single values', async () => {
+        // Add dashboard relations
+        await Promise.all(
+          dashboardItems.map((dashboardItem, i) =>
+            addRelation(dashboards[i].id, dashboardItem, i === 0 ? { test: 'yes' } : {}),
+          ),
+        );
+
+        const result = await models.dashboardRelation.findDashboardRelationsForEntityAndProject(
+          dashboardIds,
+          testEntity.code,
+          'testProject',
+        );
+
+        expect(getResultChildIds(result)).toEqual(dashboardItems.map(d => d.id));
+      });
+
+      it('should return dashboard items that match entity and project when dashboard relation filters are array values', async () => {
+        // Add dashboard relations
+        await Promise.all(
+          dashboardItems.map((dashboardItem, i) =>
+            addRelation(dashboards[i].id, dashboardItem, i === 0 ? { test: ['yes', 'no'] } : {}),
+          ),
+        );
+
+        const result = await models.dashboardRelation.findDashboardRelationsForEntityAndProject(
+          dashboardIds,
+          testEntity.code,
+          'testProject',
+        );
+
+        expect(getResultChildIds(result)).toEqual(dashboardItems.map(d => d.id));
```

Comment:
```suggestion
        expect(getResultChildIds(result).sort()).toEqual(dashboardItems.map(d => d.id).sort());
```
I was playing with the tests a bit here and noticed they were failing a bit intermittently (when I added console logs for instance). I've run into this issue a bunch of times where you do an array comparison where you don't care about the order.

I think there's an extended jest function we could add that we might use for this case, but just doing a `sort()` usually does the job